gocourse17/server/server.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/vikibell/prjctr-golang-beginning/gocourse17/service"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"log"
+	"net"
+
+	pb "github.com/vikibell/prjctr-golang-beginning/gocourse17/grpcapi"
+)
+
+type server struct {
+	pb.UnimplementedReviewServer
+
+	history service.ReviewHistory
+}
+
+func (s *server) GetHistory(_ context.Context, request *pb.GetHistoryRequest) (*pb.GetHistoryResponse, error) {
+	driverId := int(request.GetDriverId())
+
+	if driverId <= 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid driver id")
+	}
+
+	history, _ := s.history.GerReviews(driverId)
+
+	var response []*pb.ReviewData
+	for _, review := range history {
+		response = append(response, &pb.ReviewData{
+			CargoState:       pb.Rating(review.CargoState),
+			ServiceQuality:   pb.Rating(review.ServiceQuality),
+			FulfillmentSpeed: pb.Rating(review.FulfillmentSpeed),
+		})
+	}
+
+	return &pb.GetHistoryResponse{Reviews: response}, nil
+}
+
+func (s *server) SendReview(_ context.Context, request *pb.SendReviewRequest) (*pb.SendReviewResponse, error) {
+	driverId := int(request.GetDriverId())
+	if driverId <= 0 {
+		return &pb.SendReviewResponse{Message: "Fail"}, status.Errorf(codes.InvalidArgument, "invalid driver id")
```

Comment:
Не потрібно повертати ніякий респонс, якщо помилка. Вистачає:
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "invalid driver id")
```

Для помилки меседж зашифрований в `status.Errorf`.

Reply:
`Error` замість `Errorf`, бо немає форматування:
```suggestion
		return nil, status.Error(codes.InvalidArgument, "invalid driver id")
```

`Errorf` потрібно, якби:

```go
		return nil, status.Errorf(codes.InvalidArgument, "invalid driver id=%d", driverID)
```